pkg/datasource/data.go

```diff
@@ -133,16 +201,33 @@ func (ds *dataSource) Type() Type {
 	return ds.dType
 }
 
-func (ds *dataSource) NewData() Data {
+func (ds *dataSource) NewPacketSingle() PacketSingle {
 	d := &data{
-		Payload: make([][]byte, ds.payloadCount),
+		GadgetData: &api.GadgetData{
+			Data: &api.DataElement{
+				Payload: make([][]byte, ds.payloadCount),
+			},
+		},
 	}
+
+	d.dataElement = (*dataElement)(d.GadgetData.Data)
```

Comment:
Can be moved to the definition above.

Reply:
It's a pointer to what is defined above, so we first need to define it to be able to assign it here, don't we?

Reply:
🙈 You're of course right, sorry.